mesheryctl/internal/cli/root/config/config.go

```diff
@@ -444,3 +444,32 @@ func AddContextToConfig(contextName string, context Context, configPath string,
 
 	return nil
 }
+
+// UpdateContextToConfig updates the given context in meshconfig
+func UpdateContextToConfig(currCtx string, ctx *Context, configPath string) error {
+	if _, err := os.Stat(configPath); os.IsNotExist(err) {
+		return err
+	}
+
+	viper.SetConfigFile(configPath)
+	err := viper.ReadInConfig()
+	if err != nil {
+		return err
+	}
+
+	meshConfig := &MesheryCtlConfig{}
+	err = viper.Unmarshal(&meshConfig)
+	if err != nil {
+		return err
+	}
+
+	meshConfig.Contexts[currCtx] = *ctx
+	viper.Set("contexts."+currCtx, ctx)
+
+	err = viper.WriteConfig()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
I believe this logic is pre-existing in mesheryctl.

We set the central context struct and then write that context to the config file with viper.

See:

https://github.com/meshery/meshery/blob/cd022efe9fd8628541ff0a4af5b38bf5966e325d/mesheryctl/internal/cli/root/config/config.go#L58-L59

Please reuse the functions. We should only have a single way to manipulate a context, a single truth value of a context and a single way to write the context to the meshconfig file.

Reply:
@navendu-pottekkat I am replacing the func name `SetContext` with `UpdateContextToConfig` as the file follows a system of setter and getter functions where Set functions set the values of the respective variable. but `SetContext` serves a different purpose, it is to update the context within mesh-config. So renaming it would be much better to express a better understanding of the function.

Reply:
How are we consolidating updating a key in a context right now? I was under the impression that we make changes to the central context struct and write this struct as a whole to the meshconfig file. That needs to be the behaviour right?

Do we need to change anything else to get that behaviour?

Reply:
Some set function only updates values in the struct, and the changes don't reflect in meshconfig and some does reflect the changes to meshconfig.

We need to chain these functions properly, I feel a good code-cleaning is required among these functions.

I haven't included that changes in this PR though.